From 7e297c793c2bcd3418c9efeeab66b7e5656a4528 Mon Sep 17 00:00:00 2001
From: Bao Xiaowei <xiaowei.bao@nxp.com>
Date: Mon, 5 Feb 2018 16:51:35 +0800
Subject: [PATCH 409/767] misc: pci_endpoint_test: fix the call trace when the
 module removed.

Fix the call trace when call the misc_unregister function on the module
removed stage.

[recomposed the title]

Signed-off-by: Bao Xiaowei <xiaowei.bao@nxp.com>
Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/misc/pci_endpoint_test.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/pci_endpoint_test.c b/drivers/misc/pci_endpoint_test.c
index daaafd3a0efa..6e9afca9139d 100644
--- a/drivers/misc/pci_endpoint_test.c
+++ b/drivers/misc/pci_endpoint_test.c
@@ -98,6 +98,7 @@ struct pci_endpoint_test {
 	struct miscdevice miscdev;
 	enum pci_barno test_reg_bar;
 	size_t alignment;
+	char name[20];
 };
 
 struct pci_endpoint_test_data {
@@ -466,7 +467,6 @@ static int pci_endpoint_test_probe(struct pci_dev *pdev,
 	int err;
 	int irq = 0;
 	int id;
-	char name[20];
 	enum pci_barno bar;
 	void __iomem *base;
 	struct device *dev = &pdev->dev;
@@ -561,10 +561,10 @@ static int pci_endpoint_test_probe(struct pci_dev *pdev,
 		goto err_iounmap;
 	}
 
-	snprintf(name, sizeof(name), DRV_MODULE_NAME ".%d", id);
+	snprintf(test->name, sizeof(test->name), DRV_MODULE_NAME ".%d", id);
 	misc_device = &test->miscdev;
 	misc_device->minor = MISC_DYNAMIC_MINOR;
-	misc_device->name = kstrdup(name, GFP_KERNEL);
+	misc_device->name = test->name;
 	if (!misc_device->name) {
 		err = -ENOMEM;
 		goto err_ida_remove;
-- 
2.17.0

